### PR TITLE
Updates scan target schedule format

### LINKED ISSUE #59: Update scan target schedule format
We switched from cron tab to an enum format, reflect that on the SDK since the current default won't work anymore.

### PR DESCRIPTION
**Pull request by asieira**
* Creates a new enum called ScanTargetSchedule with the new possible values for scan target scan frequencies.
* Update existing client methods to accept either a string or the new enum format, performing proper input validation.
* Write tests to ensure validation and conversion of the old-style cron job format strings into the new values is working properly.
* Fixes existing tests.

Fixes #59 

Tested locally with unit tests.

### CODE REVIEW

File: zanshinsdk/test_client.py [hunk 1]
```diff
@@ -2518,3 +2518,23 @@ def test_onboard_scan_target_aws_boto3_session(self, request, mock_is_file):
...
+
+        self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value('0 * * * *'), zanshinsdk.ScanTargetSchedule.ONE_HOUR)
+        self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value('0 */6 * * *'), zanshinsdk.ScanTargetSchedule.SIX_HOURS)
+        self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value('0 */12 * * *'), zanshinsdk.ScanTargetSchedule.TWELVE_HOURS)
+        self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value('0 0 * * *'), zanshinsdk.ScanTargetSchedule.TWENTY_FOUR_HOURS)
+        self.assertEqual(zanshinsdk.ScanTargetSchedule.from_value('0 0 * * 0'), zanshinsdk.ScanTargetSchedule.SEVEN_DAYS)
+
+        self.assertRaises(TypeError, zanshinsdk.ScanTargetSchedule.from_value, 1)
+        self.assertRaises(TypeError, zanshinsdk.ScanTargetSchedule.from_value, 1.0)
+        self.assertRaises(ValueError, zanshinsdk.ScanTargetSchedule.from_value, "foo")
```
Comment (edpin) [inline zanshinsdk/test_client.py:2540]: How about checking that '0 0 */8 * *' is not acceptable, for example?

File: zanshinsdk/client.py [hunk 2]
```diff
@@ -46,6 +48,41 @@ class ScanTargetKind(str, Enum):
...
+    SIX_HOURS = '6h'
+    TWELVE_HOURS = '12h'
+    TWENTY_FOUR_HOURS = '24h'
+    SEVEN_DAYS = '7d'
+
+    @classmethod
+    def from_value(cls, schedule: Union[str, ScanTargetSchedule]) -> ScanTargetSchedule:
+        if isinstance(schedule, cls):
+            return schedule
+        elif isinstance(schedule, str):
```
Comment (edpin) [inline zanshinsdk/client.py:62]: I'd accept only the new style, but I understand not wanting to break old clients, if any.

**Review by edpin — DISMISSED** [review]

File: zanshinsdk/client.py (hunk 2 above)
Comment (asieira) [reply zanshinsdk/client.py:62]: Precisely. Don't want to break any existing customer scripts that might rely on the old syntax to work.

File: zanshinsdk/test_client.py (hunk 1 above)
Comment (asieira) [reply zanshinsdk/test_client.py:2540]: Sure, I'll add it for good measure.

**Pull request merged**